_data-prepper/pipelines/configuration/processors/aggregate.md

```diff
@@ -18,12 +18,139 @@ The following table describes the options you can use to configure the `aggregat
+{ "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "bytes": 1000 }
+{ "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "http_verb": "GET" }
+```
+
+Then, the action combines the events into one. The pipeline then uses the following combined event:
+
+```
+{ "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "status": 200, "bytes": 1000, "http_verb": "GET" }
+```
+
+### count
+
+The `count` event counts events belong to the same group and generates a new event with values of from the `identification_keys` and the count, which indicates the number of new events. You can customize the processor with the following configuration options:
+
+
+ * `count_key`: Key name to use for storing the count. Default name is `aggr._count`.
+* `start_time_key`: key name to use for storing the start time. Default name is `aggr._start_time`.
+* `output_format`: Format of the aggregated event.
+     * `otel_metrics` - Default output format. Outputs in OTel metrics SUM type with count as value
+    * `raw` - Generates a JSON object with `count_key` field with count as value and `start_time_key` field with aggregation start time as value.
+
+For an example, when using `identification_keys: ["sourceIp", "destination_ip"]`, the `count` action counts and processes the following events:
+
+```json
+{ "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "status": 200 }
+{ "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "status": 503 }
+{ "sourceIp": "127.0.0.1", "destinationIp": "192.168.0.1", "status": 400 }
+```
+
+The processor creates the following event:
+
+```json
+{"isMonotonic":true,"unit":"1","aggregationTemporality":"AGGREGATION_TEMPORALITY_DELTA","kind":"SUM","name":"count","description":"Number of events","startTime":"2022-12-02T19:29:51.245358486Z","time":"2022-12-02T19:30:15.247799684Z","value":3.0,"sourceIp":"127.0.0.1","destinationIp":"192.168.0.1"}
+```
+
+### histogram
+
+The `histogram` action aggregates event belonging to the same group and generates a new event with values of the `identification_keys` and histogram of the aggregated events based on a configured `key`. The histogram contains the number of events, sum, buckets, bucket counts, and optionally min and max of the values corresponding to the `key`. The action drops all events that make up the combined event.
+
+You can customize the processor with the following configuration options:
+
+* `key`: Name of the field in the events the histogram generates.
+* `generated_key_prefix`: `key_prefix` used by all the fields created in the aggregated event. This allows the user to make sure that the names of the histogram event does not conflict with the field names in the event
+* `units`: Name of the units for the values in the `key`.
+* `record_minmax`: A boolean indicating if the histogram should include the min and max of the values during the aggregation duration.
+* `buckets`: A list of buckets (values of type `double`) indicating the buckets in the histogram.
```

Comment:
Need a little more information. Indicating the what of the buckets in the histogram? Can it just be "A list of buckets (values of type `double`) in the histogram"?

Reply:
I'll address this in a separate PR